### PR TITLE
Np 9070 positions query

### PR DESCRIPTION
**Pull request by RanjithTom**

### CODE REVIEW

File: cristin-person/src/main/java/no/unit/nva/cristin/person/affiliations/handler/PositionCodesHandler.java [hunk 1]
```diff
@@ -35,11 +41,28 @@ public PositionCodesHandler(CristinPositionCodesClient apiClient, Environment en
...
     }
 
     @Override
     protected Integer getSuccessStatusCode(Void input, PositionCodes output) {
         return HttpURLConnection.HTTP_OK;
     }
+
+    private Boolean extractPositionCodeStatusQueryParam(RequestInfo requestInfo) throws BadRequestException {
+        var activeStatusQueryParam = requestInfo.getQueryParameterOpt(ACTIVE_STATUS_QUERY_PARAM)
+            .map(String::valueOf).orElse(null);
```
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/affiliations/handler/PositionCodesHandler.java:54]: This is already a String. I think doing `String::valueOf` is unnecessary.

Comment (RanjithTom) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/affiliations/handler/PositionCodesHandler.java:54]: This is Optional string so we need to map this to a String value

Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/affiliations/handler/PositionCodesHandler.java:54]: What about:
````
return requestInfo.getQueryParameterOpt(ACTIVE_STATUS_QUERY_PARAM)
    .map(this::validateAndGetPositionCodeStatusBooleanQueryParam).orElse(null);
````

**Review by torarnet — COMMENTED** [review]
Have commented a suggestion you should look at

File: cristin-person/src/main/java/no/unit/nva/cristin/person/affiliations/handler/PositionCodesHandler.java (hunk 1 above)
Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/affiliations/handler/PositionCodesHandler.java:54]: Optional gets unwrapped when you do the `orElse`

Comment (RanjithTom) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/affiliations/handler/PositionCodesHandler.java:54]: > What about:
> 
> ```
> return requestInfo.getQueryParameterOpt(ACTIVE_STATUS_QUERY_PARAM)
>     .map(this::validateAndGetPositionCodeStatusBooleanQueryParam).orElse(null);
> ```
validateAndGetPositionCodeStatusBooleanQueryParam throws BadRequestException so it's not able to define in method reference inside map

Will Update code as per simply using `orElse `for unwrapping string

**Review by torarnet — DISMISSED** [review]

**Review by axthosarouris — CHANGES_REQUESTED** [review]
Good job. Replace with var where applicable

**Pull request merged**